src/fileAttachment.js

```diff
@@ -70,6 +71,10 @@ class AbstractFile {
   async html() {
     return this.xml("text/html");
   }
+  async xlsx() {
+    const [ExcelJS, buffer] = await Promise.all([requireDefault(exceljs.resolve()), this.arrayBuffer()]);
+    return new ExcelWorkbook(await new ExcelJS.Workbook().xlsx.load(buffer));
```

Comment:
I’m not sure if there’s really an advantage to this, but it _is_ possible to stream…

```suggestion
    const [ExcelJS, stream] = await Promise.all([requireDefault(exceljs.resolve()), this.stream()]);
    return new ExcelWorkbook(await new ExcelJS.Workbook().xlsx.read(readStream(stream)));
```

where

```js
async function* readStream(stream) {
  const reader = stream.getReader();
  try {
    while (true) {
      const {done, value} = await reader.read();
      if (done) return;
      yield value;
    }
  } finally {
    reader.releaseLock();
  }
}
```

That said, it feels like purely a novelty, so unless you know of a reason to favor streaming I would stick with this approach and just leave this comment for history in case we’d like to revisit.

Reply:
This might be better on memory usage for large files, but unknown how much better. It'd be interesting to find some realistically large files and see what performance is really like.